service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java

```diff
@@ -0,0 +1,462 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.uprotocol.core.udiscovery;
+
+import static org.eclipse.uprotocol.common.util.UStatusUtils.buildStatus;
+import static org.eclipse.uprotocol.common.util.UStatusUtils.checkArgument;
+import static org.eclipse.uprotocol.common.util.UStatusUtils.isOk;
+import static org.eclipse.uprotocol.common.util.UStatusUtils.toStatus;
+import static org.eclipse.uprotocol.common.util.log.Formatter.join;
+import static org.eclipse.uprotocol.common.util.log.Formatter.quote;
+import static org.eclipse.uprotocol.common.util.log.Formatter.stringify;
+import static org.eclipse.uprotocol.common.util.log.Formatter.tag;
+import static org.eclipse.uprotocol.core.udiscovery.common.Constants.TOPIC_NODE_NOTIFICATION;
+import static org.eclipse.uprotocol.core.udiscovery.internal.Utils.logStatus;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_ADD_NODES;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_DELETE_NODES;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_FIND_NODES;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_FIND_NODE_PROPERTIES;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_LOOKUP_URI;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_REGISTER_FOR_NOTIFICATIONS;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_UNREGISTER_FOR_NOTIFICATIONS;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_UPDATE_NODE;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.METHOD_UPDATE_PROPERTY;
+import static org.eclipse.uprotocol.core.udiscovery.v3.UDiscovery.SERVICE;
+import static org.eclipse.uprotocol.transport.builder.UPayloadBuilder.packToAny;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import org.eclipse.uprotocol.UPClient;
+import org.eclipse.uprotocol.common.UStatusException;
+import org.eclipse.uprotocol.common.util.log.Key;
+import org.eclipse.uprotocol.core.udiscovery.db.DiscoveryManager;
+import org.eclipse.uprotocol.core.udiscovery.v3.FindNodePropertiesResponse;
+import org.eclipse.uprotocol.core.udiscovery.v3.FindNodesResponse;
+import org.eclipse.uprotocol.core.udiscovery.v3.LookupUriResponse;
+import org.eclipse.uprotocol.rpc.CallOptions;
+import org.eclipse.uprotocol.rpc.URpcListener;
+import org.eclipse.uprotocol.uri.builder.UResourceBuilder;
```

Comment:
the package was renamed to factory per request from Mikael

Reply:
Done